src/eth_provider/database/types/header.rs

```diff
@@ -1,44 +1,36 @@
+#[cfg(any(test, feature = "arbitrary", feature = "testing"))]
+use {
+    arbitrary::Arbitrary,
+    reth_primitives::{constants::EMPTY_ROOT_HASH, B256, B64, U256},
+};
 
 /// A header as stored in the database
 #[derive(Debug, Serialize, Deserialize, Hash, Clone, PartialEq, Eq)]
+#[cfg_attr(any(test, feature = "arbitrary", feature = "testing"), derive(arbitrary::Arbitrary))]
 pub struct StoredHeader {
     #[serde(deserialize_with = "crate::eth_provider::database::types::serde::deserialize_intermediate")]
     pub header: Header,
 }
 
 #[cfg(any(test, feature = "arbitrary", feature = "testing"))]
-impl<'a> arbitrary::Arbitrary<'a> for StoredHeader {
-    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
-        let header = SealedHeader::arbitrary(u)?;
-
-        Ok(StoredHeader {
+impl<'a> StoredHeader {
+    pub fn arbitrary_with_optional_fields(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(Self {
             header: Header {
-                hash: Some(header.hash()),
-                parent_hash: header.parent_hash,
-                uncles_hash: header.ommers_hash,
-                miner: header.beneficiary,
-                state_root: header.state_root,
-                transactions_root: header.transactions_root,
-                receipts_root: header.receipts_root,
-                logs_bloom: header.logs_bloom,
-                difficulty: header.difficulty,
-                number: Some(header.number),
-                gas_limit: u128::from(header.gas_limit),
-                gas_used: u128::from(header.gas_used),
-                timestamp: header.timestamp,
-                total_difficulty: Some(U256::arbitrary(u)?),
-                extra_data: header.extra_data.clone(),
-                mix_hash: Some(header.mix_hash),
-                nonce: Some(B64::from(header.nonce)),
-                base_fee_per_gas: header.base_fee_per_gas.map(|base_fee_per_gas| base_fee_per_gas as u128),
+                hash: Some(B256::arbitrary(u)?),
+                total_difficulty: Some(U256::arbitrary(u).unwrap()),
+                mix_hash: Some(B256::arbitrary(u).unwrap()),
+                nonce: Some(B64::arbitrary(u).unwrap()),
                 withdrawals_root: Some(EMPTY_ROOT_HASH),
-                blob_gas_used: header.blob_gas_used.map(|blob_gas_used| blob_gas_used as u128),
-                excess_blob_gas: header.excess_blob_gas.map(|excess_blob_gas| excess_blob_gas as u128),
-                parent_beacon_block_root: header.parent_beacon_block_root,
+                base_fee_per_gas: Some(u64::arbitrary(u).unwrap() as u128),
+                blob_gas_used: Some(u64::arbitrary(u).unwrap() as u128),
+                excess_blob_gas: Some(u64::arbitrary(u).unwrap() as u128),
+                gas_limit: u64::arbitrary(u).unwrap() as u128,
+                gas_used: u64::arbitrary(u).unwrap() as u128,
+                number: Some(u64::arbitrary(u).unwrap()),
```

Comment:
why can't we use `..Self::arbitrary(u)?.header` for the rest of these fields?

Reply:
In fact all the fields that are put here outside the `..Self::arbitrary(u)?.header` have two particular reasons:
- Either it is fields that will have a problem during conversion (for example `gas_limit` must be `u64` when converting to primitive even if it is a `u128` in the rpc type).
- Either these are fields which are optional but which need to have a `Some` when converting to primitive. As the `..Self::arbitrary(u)?.header` allows an `Option` to be `None`, this would give us systematic errors during conversions on the test suite...